File: WIP - Progetto/app/src/main/java/it/wip/database/model/ShopElement.kt
package it.wip.database.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "shop_element")
data class ShopElement(
    @PrimaryKey
    @ColumnInfo(name = "element_name")
    val elementName: String,
    val type: String,
    val description: String,
    val price: Int
)

File: wip_flutter/lib/database/model/shop_element.dart
class ShopElement {

  final String elementName;
  final String type;

  ShopElement({
    required this.elementName,
    required this.type,
  });

  Map<String, dynamic> toMap() {
    return {
      'element_name': elementName,
      'type': type
    };
  }

}